doc/src/cylc-user-guide/cug.tex

```diff
@@ -6166,6 +6092,466 @@ \subsubsection{Where To Put Batch System Handler Modules}
+\lstset{language=suiterc}
+\begin{lstlisting}
+    [[xtriggers]]
+        clock_1 = wall_clock(PT1H)
+\end{lstlisting}
+
+Finally, a zero-offset clock trigger does not need to be declared under
+the \lstinline=[xtriggers]= section:
+\lstset{language=suiterc}
+\begin{lstlisting}
+[scheduling]
+    initial cycle point = 2018-01-01
+    [[dependencies]]
+        [[[P1D]]]
+            # zero-offset clock trigger:
+            graph = "@wall_clock => foo"
+[runtime]
+    [[foo]]
+        script = run-foo.sh
+\end{lstlisting}
+
+\subsection{Built-in Suite State Triggers}
+\label{Built-in Suite State Triggers}
+
+These can be used instead of the older suite state polling tasks described
+in~\ref{SuiteStatePolling} for inter-suite triggering - i.e.\ to trigger local
+tasks off of remote task statuses or messages in other suites.
+
+The suite state trigger function signature looks like this:
+\begin{lstlisting}
+suite_state(suite, task, point, offset=None, status='succeeded',
+            message=None, cylc_run_dir=None, debug=False)
+\end{lstlisting}
+
+The first three arguments are compulsory; they single out the target suite name
+(\lstinline=suite=) task name (\lstinline=task=) and cycle point
+(\lstinline=point=). The function argments mirror the arguments and options of
+the \lstinline=cylc suite-state= command - see
+\lstinline=cylc suite-state --help= for documentation.
+
+As a simple example, consider the suites in
+\lstinline=<cylc-dir>/etc/dev-suites/xtrigger/suite_state/=. The ``upstream''
+suite (which we want to trigger off of) looks like this:
+
+\lstinputlisting{../../../etc/dev-suites/xtrigger/suite_state/upstream/suite.rc}
+
+It must be registered and run under the name {\em up}, as referenced in the
+``downstream'' suite that depends on it:
+
+\lstinputlisting{../../../etc/dev-suites/xtrigger/suite_state/downstream/suite.rc}
+
+Try starting the downstream suite first, then the upstream, and watch what happens.
+In each cycle point the \lstinline=@upstream= trigger in the downstream suite
+waits on the task \lstinline=foo= (with the same cycle point) in the upstream
+suite to emit the {\em data ready} message. 
+
+Some important points to note about this:
+\begin{myitemize}
+  \item  the function call interval, which determines how often the suite
+    server program checks the clock, is optional. Here it is
+    \lstinline=PT10S= (i.e.\ 10 seconds, which is also the default value).
+  \item the \lstinline=suite_state= trigger function, like the
+    \lstinline=cylc suite-state= command, must have read-access to the upstream
+    suite's public database. 
+  \item the cycle point argument is supplied by a string template
+    \lstinline=%(point)s=. The string templates available to trigger function
+    arguments are described in {\em Custom Trigger Functions} (\ref{Custom
+    Trigger Functions}).
+\end{myitemize}
+
+The return value of the \lstinline=suite_state= trigger function looks like
+this:
+\begin{lstlisting}
+    results = {
```

Comment:
It can handle this already (if I understand your point correctly) because we split AND expressions into separate triggers.  It's just conditional OR expressions that are currently off-limits for external triggers.